### PR TITLE
Attempt to embed Node in Orogene

### PR DESCRIPTION
**Pull request by jlkiri**
This PR creates a new crate `obelisk` that has Node as its git submodule.
Custom C API is used to call Node from Rust.
Node is checked out at version 14.14.0.

Before we make some automation, build needs some manual steps.
Because of that, CI build will fail and that's why I commented out the relevant lines.

### CODE REVIEW

**Review by zkat — APPROVED** [review]

File: crates/obelisk/node_c_api/node_ext_api.cc [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    std::unique_ptr<Environment, decltype(&node::FreeEnvironment)> env(
+        node::CreateEnvironment(isolate_data.get(), context, args, exec_args),
+        node::FreeEnvironment);
+
+    MaybeLocal<Value> loadenv_ret = node::LoadEnvironment(
+        env.get(),
+        "const publicRequire ="
+        "  require('module').createRequire(process.cwd() + '/');"
+        "globalThis.require = publicRequire;"
+        "globalThis.embedVars = { nön_ascıı: '🏳️‍🌈' };"
```
Comment (addaleax) [inline crates/obelisk/node_c_api/node_ext_api.cc:105]: Love this line. :+1: 

In all seriousness, I just wanted to say hi! It’s cool that you’re using this and that if you need **anything** regarding the Node.js embedder API, please please please ping me. :slightly_smiling_face:

Comment (jlkiri) [reply crates/obelisk/node_c_api/node_ext_api.cc:105]: Hi! Yeah, when time comes to customize this code I would love some help 🙂

File: crates/obelisk/build.rs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        println!("cargo:rustc-link-lib=static=histogram");
+        println!("cargo:rustc-link-lib=static=llhttp");
+        println!("cargo:rustc-link-lib=static=nghttp2");
+        println!("cargo:rustc-link-lib=static=openssl");
+        println!("cargo:rustc-link-lib=static=torque_base");
+
+        cc::Build::new()
+            .cpp(true)
+            .file("node.cpp")
+            .compile("liboronode.a"); */
```
Comment (zkat) [inline crates/obelisk/build.rs:51]: I assume you want to uncomment this?

File: crates/obelisk/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,19 @@
+use std::os::raw::c_char;
+
+extern "C" {
+    pub fn run_node(code: *const c_char) -> i32;
```
Comment (zkat) [inline crates/obelisk/src/lib.rs:4]: Is there gonna be a version of this that works with file paths instead, rather than a string of code?

**Review by zkat — COMMENTED** [review]
I'm very excited about this! Thanks for taking this on!

File: crates/obelisk/src/lib.rs (hunk 3 above)
Comment (jlkiri) [reply crates/obelisk/src/lib.rs:4]: Yes! There should at least be a way to run a file! The official examples only have an analogy of `node -e` - that is something to work on.

**Pull request merged**